scrappers/tbib/lib/Tbib.rb

```diff
@@ -0,0 +1,76 @@
+class Tbib
+	include RethinkDB::Shortcuts
+
+	def initialize()
+		@conn = r.connect(:host => $config.server_ip, :port => $config.server_port)
+		$log.log 'connected to database!'
+		@conn.use($config.database)
+		@mutex = Mutex.new
+		if !r.table_list.run(@conn).include?($config.table)
+			r.table_create($config.table).run(@conn)
+			$log.log "Created table #{$config.table} at #{$config.db}."
+		else
+			$log.log "Table #{$config.table} already exists."
+		end
+	rescue => e
+		$log.fatal("Could not connect to database on #{$config.server_ip}:#{$config.server_port}!")
+		$log.fatal(e)
+		exit 1
+	end
+
+	def start!
+		url2 = "https://tbib.org/index.php?page=dapi&s=post&q=index&tags=#{$config.tags.join('+')}"
+		$log.log "Connecting to tbib..."
+		xml = open(url2).read
+		c = Nokogiri.XML(xml).xpath('//posts').first['count'].to_i
+		$log.log "Total number of images is: #{c}"
+		if c == 0
+			$log.error 'No results.'
+			exit 0
+		end
+		images = []
+		procs = []
+		(0..(c/100 + (c%100 == 0 ? 0 : 1))).to_a.each do |pid|
+			procs << proc {
+				Thread.new do
+					url = "https://tbib.org/index.php?page=dapi&s=post&q=index&tags=#{$config.tags.join('+')}&json=1&pid=#{pid}"
+					j = JSON.parse(open(url).read)
+					@mutex.synchronize do
```

Comment:
Should probably move that uo a little, so it'd also handle the request pausing (and also change that to a semaphore, as stated previously)